pkg/services/ngalert/alert_definition.go

```diff
@@ -0,0 +1,78 @@
+package ngalert
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana/pkg/services/ngalert/eval"
+)
+
+// GetEvalCondition returns a condition suitable for execution
+// based on the saved AlertDefinition.
+func (alertDefinition *AlertDefinition) GetEvalCondition(now time.Time) (*eval.Condition, error) {
```

Comment:
Doesn't have to be public:
```suggestion
// getEvalCondition returns a condition suitable for execution
// based on the saved AlertDefinition.
func (alertDefinition *AlertDefinition) getEvalCondition(now time.Time) (*eval.Condition, error) {
```